llvm/lib/Transforms/Vectorize/VPlanAnalysis.cpp

```diff
@@ -233,6 +233,11 @@ Type *VPTypeAnalysis::inferScalarTypeForRecipe(const VPReplicateRecipe *R) {
   llvm_unreachable("Unhandled opcode");
 }
 
+Type *
+VPTypeAnalysis::inferScalarTypeForRecipe(const VPPartialReductionRecipe *R) {
+  return R->getUnderlyingInstr()->getType();
```

Comment:
Can the type be inferred from the operands?

Reply:
I think it would just be moving `VPPartialReductionRecipe` into the `.Case` statement above the current one (alongside `VPReductionRecipe`) in the base `inferScalarType(const VPValue *)`, which will recursively infer the scalar type based on the first operand. Then you wouldn't need this extra method.